xdelta3-dir-patcher

```diff
@@ -174,21 +607,57 @@ class XDelta3DirPatcher(object):
 
         print("Done")
 
-    # TODO: Unit test me
-    def apply(self, old_dir, patch_bundle, target_dir, root_patch_dir = None):
-        # Create a temp dir
-        patch_dir = mkdtemp(prefix="%s_" % self.__class__.__name__)
-        print("Using %s as staging area" % patch_dir)
-
-        print("Extracting archive...")
-        with tarfile.open(patch_bundle) as patch_archive:
-            patch_archive.extractall(patch_dir)
+    # XXX This implementation requires directories passed in to be
+    #     empty. Callers are required to send in files first if they
+    #     expect a mixture of dirs/files of the same path.
+    @staticmethod
+    def remove_item(target_dir, deleted_item, debug = False, attempt = 0):
+        deleted_item_path = path.join(target_dir, deleted_item)
+        if debug:
+            print('Deleting %s' % deleted_item_path)
+        else:
+            print('X', end = "")
+
+        if not path.exists(deleted_item_path):
+            return
```

Comment:
@sgnn7 Why would an item be remove twice? I would think this should be an exception if the path doesn't exist.